pyhap/accessory_driver.py

```diff
@@ -86,9 +87,15 @@ def __init__(self, accessory, state):
         self.state = state
 
         adv_data = self._get_advert_data()
+        # Append part of MAC address to prevent name conflicts
+        name = '{} {}.{}'.format(
+            self.accessory.display_name,
+            self.state.mac[-8:].replace(':', ''),
+            HAP_SERVICE_TYPE
+        )
         super().__init__(
-            '_hap._tcp.local.',
-            name=self.accessory.display_name + '._hap._tcp.local.',
+            HAP_SERVICE_TYPE,
+            name=name,
```

Comment:
Will this break existing pairings that are looking for the old name ?

Reply:
When I tested it, I had already paired an accessory and then changed the name and it remained paired in Home. I'm guessing that because the data (the ID?) in the advertisement payload isn't changed, HomeKit knows it's the same device and keeps it.

Reply:
I'll test this later today to make sure its not going to cause anything fallout with Home Assistant

Reply:
It looks like we limit the name length in Home Assistant to 25 chars 
https://github.com/home-assistant/core/pull/15600

That should ensure we don't hit the mdns label max length of 63 chars